File: summer/WebContent/js/system/componentmodel/ICollectionViewLiveShaping.js
/**
 * ICollectionViewLiveShaping
 */

define(["dojo/_base/declare", "system/Type"], function(declare, Type){
	var ICollectionViewLiveShaping = declare("EntryIndex", null,{
	});
	
	Object.defineProperties(ICollectionViewLiveShaping.prototype,{

//		bool 
		CanChangeLiveSorting:
		{
			get:function(){}
		},
//		bool 
		CanChangeLiveFiltering:
		{
			get:function(){}
		},
//		bool 
		CanChangeLiveGrouping:
		{
			get:function(){}
		},
//		bool? 
		IsLiveSorting:
		{
			get:function(){},
			set:function(value){}
		},
//		bool? 
		IsLiveFiltering:
		{
			get:function(){},
			set:function(value){}
		},
//		bool? 
		IsLiveGrouping:
		{
			get:function(){},
			set:function(value){}
		},
//		ObservableCollection<string> 
		LiveSortingProperties:
		{
			get:function(){}
		},
//		ObservableCollection<string> 
		LiveFilteringProperties:
		{
			get:function(){}
		},
//		ObservableCollection<string> 
		LiveGroupingProperties:
		{
			get:function(){}
		},
	});
	
	ICollectionViewLiveShaping.Type = new Type("ICollectionViewLiveShaping", ICollectionViewLiveShaping, [Object.Type]);
	return ICollectionViewLiveShaping;
});

File: org.summer.view.widget/src/org/summer/view/widget/model/ICollectionViewLiveShaping.java
package org.summer.view.widget.model;

import org.summer.view.widget.collection.ObservableCollection;

public interface ICollectionViewLiveShaping{
	boolean CanChangeLiveFiltering { get; }
	boolean CanChangeLiveGrouping { get; }
	boolean CanChangeLiveSorting { get; }
	boolean IsLiveFiltering { get; set; }
	boolean IsLiveGrouping { get; set; }
	boolean IsLiveSorting { get; set; }
	ObservableCollection<String> LiveFilteringProperties { get; }
	ObservableCollection<String> LiveGroupingProperties { get; }
	ObservableCollection<String> LiveSortingProperties { get; }
}
